cmd/micro/micro.go

```diff
@@ -466,15 +468,13 @@ func DoEvent() {
 			os.Exit(0)
 		}
 		return
+	default:
+		ulua.Lock.Lock()
+		if action.InfoBar.HasPrompt {
+			action.InfoBar.HandleEvent(event)
+		} else {
+			action.Tabs.HandleEvent(event)
+		}
+		ulua.Lock.Unlock()
 	}
-
-	ulua.Lock.Lock()
-	// if event != nil {
```

Comment:
Why not just uncomment this check? (And probably also move it above the lock, no point in locking if the event is nil.)

Reply:
Or, for example, just add
```
if event == nil {
	return
}
```
before the `tcell.EventError` check. That also seems cleaner that `case nil` etc.

Reply:
> Why not just uncomment this check?

Because I would like to combine it with the `tcell.EventError` checks as well, because all of them follow the `event` "topic". At least from my perspective it looks "cleaner". And `event` is already `nil` while checking for `tcell.EventError`, but doesn't fail due to the type check capabilities of Go.

> Or, for example, just add [...] before the `tcell.EventError` check. That also seems cleaner that `case nil` etc.

Looking at `DoEvent()` alone this would mean leaving it and maybe assuming to "drop" the redraw event. Yes, it's called in a loop, but from function perspective this would be implicit instead of explicit.

Reply:
Ok, maybe it's a matter of preference. But if you insist, then the above `return` (before the `default` case) becomes unneeded. Also, maybe it makes sense to add a comment like `// no event from tcell` after `case nil:`, for slightly better readability.

Reply:
You're right, the `return` then needs to be removed to keep my argument valid. :+1: